kpi-tracker/src/main.rs

```diff
@@ -920,34 +976,117 @@ async fn db_insert_block(db: &mut DBConn, block_data: BlockData) -> anyhow::Resu
         .commit()
         .await
         .context("Failed to commit DB transaction.")?;
-    Ok(())
+
+    let end = chrono::Utc::now().signed_duration_since(start);
+    Ok((height, end))
 }
 
 /// Runs a process of inserting data coming in on `block_receiver` in a database
 /// defined in `db_connection`
 async fn run_db_process(
     db_connection: tokio_postgres::config::Config,
     mut block_receiver: tokio::sync::mpsc::Receiver<BlockData>,
-    height_sender: tokio::sync::oneshot::Sender<AbsoluteBlockHeight>,
+    height_sender: tokio::sync::oneshot::Sender<Option<AbsoluteBlockHeight>>,
+    stop_flag: Arc<AtomicBool>,
 ) -> anyhow::Result<()> {
-    let mut db = DBConn::create(db_connection, true).await?;
+    let mut db = DBConn::create(db_connection.clone(), true).await?;
     let latest_height = db
         .prepared
         .get_latest_height(&db.client)
         .await
         .context("Could not get best height from database")?
-        .map_or(0.into(), |h| h.next());
+        .or_else(|| Some(0.into()));
```

Comment:
When is `None` used then?

Reply:
Well, the intention was to use `None` as a way to communicate "start from 0", as otherwise the height returned from this is interpreted as "latest height recorded in db", thus we should use `.next()` of the height as start of the block stream.

The current implementation is wrong though, as `None` never passes through. This will be fixed. Good catch.